src/eosjs-api.ts

```diff
@@ -260,13 +260,15 @@ export class Api {
      * Named Parameters:
      *    * `broadcast`: broadcast this transaction?
      *    * `sign`: sign this transaction?
+     *    * `compression`: compress this transaction?
      *    * If both `blocksBehind` and `expireSeconds` are present,
      *      then fetch the block which is `blocksBehind` behind head block,
      *      use it as a reference for TAPoS, and expire the transaction `expireSeconds` after that block's time.
      * @returns node response if `broadcast`, `{signatures, serializedTransaction}` if `!broadcast`
      */
-    public async transact(transaction: any, { broadcast = true, sign = true, blocksBehind, expireSeconds }:
-        { broadcast?: boolean; sign?: boolean; blocksBehind?: number; expireSeconds?: number; } = {}): Promise<any> {
+    public async transact(transaction: any, { broadcast = true, sign = true, compression, blocksBehind, expireSeconds }:
```

Comment:
It may be time to refactor this out into a `TransactParams` type, it's getting a bit large.

Reply:
Or actually I think it would be `TransactArgs`, I think that's the pattern elsewhere.

Reply:
I used TransactConfig in https://github.com/EOSIO/eosjs/pull/677

Reply:
AH right, that works 👍